src/Nethermind/Nethermind.Synchronization/Peers/ISyncPeerPool.cs

```diff
@@ -105,10 +105,18 @@ public interface ISyncPeerPool : IDisposable
 
     public static class SyncPeerPoolExtensions
     {
-        public static async Task<T> AllocateAndRun<T>(this ISyncPeerPool syncPeerPool, Func<ISyncPeer, Task<T>> func, IPeerAllocationStrategy peerAllocationStrategy, AllocationContexts allocationContexts,
+        public static async Task<T> AllocateAndRun<T>(
+            this ISyncPeerPool syncPeerPool,
+            Func<ISyncPeer, Task<T>> func,
+            IPeerAllocationStrategy peerAllocationStrategy,
+            AllocationContexts allocationContexts,
             CancellationToken cancellationToken)
         {
-            SyncPeerAllocation? allocation = await syncPeerPool.Allocate(peerAllocationStrategy, allocationContexts, cancellationToken: cancellationToken);
+            SyncPeerAllocation? allocation = await syncPeerPool.Allocate(
+                peerAllocationStrategy,
+                allocationContexts,
+                timeoutMilliseconds: int.MaxValue,
```

Comment:
Isn't it a bit risky? If someone doesn't pass a cancellation token to this helper method, it will never timeout?

Also when using it in FCU we have 2 seconds hardcoded - shouldn't we use some static from `Timeouts` class?

Reply:
IMO its risky to use `CancellationToken.None`. Having a predefined timeout is harder as you'd never know the appropriate timeout. It make more sense in `FetchHeaderFromPeer` though as it never retry if header is null so user will need to handle retry themself anyway.

Reply:
For example, so now I'm trying to add the timeout in `Timeouts`. But the 2 second is kinda only relevant for FCU since it kinda need to return something in a reasonable amount of time for a REST api, so 2 second is a bit generous there. The default Eth timeout is 10 second, so if the requestor is long running, like in sync, might as well set it to 10 second. In any case, header is small, so 2 second in general is probably fine, but only for header.